### PR TITLE
Pg 3565 add field placeholders

### PR DESCRIPTION
**Pull request by snake14**
### Description:

Product wanted placeholders added to all free-form inputs so that the label is always above the input. The PR does that.

Since I was already editing the necessary Vue templates for a couple other tickets, I implemented those as part of this PR as well.

Full changes:
* Rename description fields to: Description (optional)
* Add placeholders to all text inputs so that the label is always above the field
* Move the Trigger/Variable type up next to the form title instead of being a disabled field
* Redirect to the Tag/Trigger/Variable list after a Tag/Trigger/Variable is created/updated

Internal tickets:
* PG-3565
* PG-3686
* PG-3347

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by AltamashShaikh — APPROVED** [review]
Changes looks good to me and the redirect after save also works as expected :+1:

**Comment by adamreng** [comment]
Maybe missing a placeholder on this one under the "Show advanced settings" in VariableEdit.vue:

https://github.com/matomo-org/tag-manager/blob/4b989af8b0e2bb288b4e21a86363092d16f2a88f/vue/src/Variable/VariableEdit.vue#L99-L106

File: vue/src/Tag/TagEdit.vue [hunk 1]
```diff
@@ -693,13 +697,9 @@ export default defineComponent({
...
 
-        const idTag = response.value;
-
         TagsStore.reload(this.idContainer, this.idContainerVersion).then(() => {
-          MatomoUrl.updateHash({
-            ...MatomoUrl.hashParsed.value,
-            idTag,
-          });
+          // Go back to the list of tags
+          this.cancel();
```
Comment (araichyk) [inline vue/src/Tag/TagEdit.vue:702]: Non-blocking question: is there a reason we can't this.cancel() without even calling the reload?

File: vue/src/Tag/TagEdit.vue [hunk 2]
```diff
@@ -693,13 +697,9 @@ export default defineComponent({
...
```
Comment (adamreng) [inline vue/src/Tag/TagEdit.vue:703]: Non-blocking question: is there a reason we can't this.cancel() without even calling the reload?

**Review by adamreng — APPROVED** [review]
Just found 1 missing placeholder, but aside from that, it all seems to work as expected. Very nice 👍

**Comment by snake14** [comment]
> Maybe missing a placeholder on this one under the "Show advanced settings" in VariableEdit.vue:
> 
> https://github.com/matomo-org/tag-manager/blob/4b989af8b0e2bb288b4e21a86363092d16f2a88f/vue/src/Variable/VariableEdit.vue#L99-L106

@araichyk Good catch. I went ahead and added one :+1:

**Pull request merged**